torch/_inductor/dependencies.py

```diff
@@ -591,9 +596,25 @@ def extract_read_writes(
             )
         for entry in fn.memory_usage[MemoryUsageType.INDEX_EXPR]:
             inner.index_expr(name_to_index[entry.index_name], None)
-        for entry in fn.memory_usage[MemoryUsageType.BUCKETIZE]:
+
+        # At Python 3.12, this can be replaced by itertools.batched.  We need this
```

Comment:
Actually this should be replace by itertools pairwise in Python 3.10, no?

Reply:
@Skylion007 `pairwise` is not actually what we want here. For an iterable `A, B, C, D`, I want `(A, B), (C, D)`. `pairwise` returns `(A, B), (B, C), (C, D)`, while `batched` can be configured to return `(A, B), (C, D)`.